ui/components/src/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,100 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react';
+import {
+  Button,
+  ButtonProps,
+  Dialog as MuiDialog,
+  DialogActions,
+  DialogContent,
+  DialogContentProps as MuiDialogContentProps,
+  DialogProps,
+  DialogTitle,
+  DialogTitleProps,
+  IconButton,
+  Theme,
+} from '@mui/material';
+import CloseIcon from 'mdi-material-ui/Close';
+import { combineSx } from '../utils';
+
+export interface DialogHeaderProps extends DialogTitleProps {
+  /**
+   * Callback fired when close button is clicked. If undefined, close button will not appear in header.
+   */
+  onClose?: (e: React.MouseEvent<HTMLElement>) => void;
+}
+
+export interface DialogButtonProps extends Omit<ButtonProps, 'variant' | 'color'> {
```

Comment:
If you don't put ``type`` in the excluded properties, somebody using it will have some issues no ?